### PR TITLE
chore: update assume-no-moving-gc to fix go 1.20

### LINKED ISSUE #8399: logcli 2.7.3 failed to run with go1.20
👋 while [regression testing go1.20 upgrade](https://github.com/Homebrew/homebrew-core/pull/122082), I ran into the following error:

```
level=info ts=2023-02-02T03:58:16.722819Z caller=loki.go:461 msg="Loki started"
==> /opt/homebrew/Cellar/logcli/2.7.3/bin/logcli --addr=http://localhost:62567 labels
panic: Something in this program imports go4.org/unsafe/assume-no-moving-gc to declare that it assumes a non-moving garbage collector, but your version of go4.org/unsafe/assume-no-moving-gc hasn't been updated to assert that it's safe against the go1.20 runtime. If you want to risk it, run with environment variable ASSUME_NO_MOVING_GC_UNSAFE_RISK_IT_WITH=go1.20 set. Notably, if go1.20 adds a moving garbage collector, this program is unsafe to use.

goroutine 1 [running]:
go4.org/unsafe/assume-no-moving-gc.init.0()
	go4.org/unsafe/assume-no-moving-gc@v0.0.0-20220617031537-928513b29760/untested.go:25 +0x178
```

full build log, https://pipelines.actions.githubusercontent.com/serviceHosts/d065b819-d194-4baa-bc84-7b4a294d8693/_apis/pipelines/1/runs/564571/signedlogcontent/19?urlExpires=2023-02-03T17%3A06%3A38.0355252Z&urlSigningMethod=HMACV1&urlSignature=pSX%2FY%2FAHvb9NFrLSm1xQHLgwZ%2FwQ5NinBMs1z6lh5sI%3D

### PR DESCRIPTION
**Pull request by paper42**
**What this PR does / why we need it**:
assume-no-moving-gc makes the program exit if the current go version was not explicitly specified in it as safe. This version adds go 1.20 to the list of safe versions which fixes running loki when built with go 1.20.

**Which issue(s) this PR fixes**:
Fixes #8399

**Special notes for your reviewer**:
I am not sure where I should add this change in the changelog, this seems like something that should be cherry picked to a new 2.7 patch release, but the top of the changelog seems to be the next minor version. Should I open a separate PR for the next minor stable version with similar changes? Which branch should that target?

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: operator/go.mod [hunk 1]
```diff
@@ -146,7 +146,7 @@ require (
 	go.uber.org/multierr v1.7.0 // indirect
 	go.uber.org/zap v1.21.0 // indirect
 	go4.org/intern v0.0.0-20211027215823-ae77deb06f29 // indirect
-	go4.org/unsafe/assume-no-moving-gc v0.0.0-20220617031537-928513b29760 // indirect
+	go4.org/unsafe/assume-no-moving-gc v0.0.0-20230209150437-ee73d164e760 // indirect
```
Comment (periklis) [inline operator/go.mod:149]: Would you mind to add this change in a separate PR please? The operator subproject is following it's own golang cadence.

Comment (paper42) [reply operator/go.mod]: split to #8507

**Comment by chaudum** [comment]
> I am not sure where I should add this change in the changelog, this seems like something that should be cherry picked to a new 2.7 patch release, but the top of the changelog seems to be the next minor version. Should I open a separate PR for the next minor stable version with similar changes? Which branch should that target?

I think you can put a changelog entry underneath the `Build` section of `Main/Unreleased`.

Regarding the backport: I'm not sure if we target another 2.7.x release, but you can make a PR against the `release-2.7.x` branch with the dependency update.

**Comment by tristanmorgan** [comment]
Should the single file pkg/logql/log/ip.go be updated to removed the dependancy on inet.af/netaddr so this package (go4.org/unsafe/assume-no-moving-gc) is no longer needed?

**Comment by chaudum** [comment]
@paper42 This is superseded by #8560

**Pull request closed without merging**